bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/SonnenHandler.java

```diff
@@ -0,0 +1,276 @@
+                updateStatus(ThingStatus.ONLINE);
+                updateLinkedChannels();
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message.getStatusDesription());
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message.getStatusDesription());
+        }
+    }
+
+    /**
+     * Calls the service to update the battery data
+     *
+     * @param postdata
+     */
+    private boolean updatebatteryData(@Nullable String postdata) {
+        Helper message = new Helper();
+        if (serviceCommunication.refreshBatteryConnection(message, this.getThing().getUID().toString())) {
+            updateStatus(ThingStatus.ONLINE);
+            return true;
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    message.getStatusDesription());
+            return false;
+        }
+    }
+
+    private void updateLinkedChannels() {
+        verifyLinkedChannel(CHANNELBATTERYCHARGING);
+        verifyLinkedChannel(CHANNELBATTERYDISCHARGING);
+        verifyLinkedChannel(CHANNELGRIDFEEDIN);
+        verifyLinkedChannel(CHANNELCONSUMPTION);
+        verifyLinkedChannel(CHANNELSOLARPRODUCTION);
+        verifyLinkedChannel(CHANNELBATTERYLEVEL);
+        verifyLinkedChannel(CHANNELFLOWCONSUMPTIONBATTERY);
+        verifyLinkedChannel(CHANNELFLOWCONSUMPTIONGRID);
+        verifyLinkedChannel(CHANNELFLOWCONSUMPTIONPRODUCTION);
+        verifyLinkedChannel(CHANNELFLOWGRIDBATTERY);
+        verifyLinkedChannel(CHANNELFLOWPRODUCTIONBATTERY);
+        verifyLinkedChannel(CHANNELFLOWPRODUCTIONGRID);
+
+        if (!linkedChannels.isEmpty()) {
+            updatebatteryData(null);
+            for (Channel channel : getThing().getChannels()) {
+                updateChannel(channel.getUID().getId());
+            }
+            startAutomaticRefresh();
+            automaticRefreshing = true;
+        }
+    }
+
+    private void verifyLinkedChannel(String channelID) {
+        if (isLinked(channelID) && !linkedChannels.containsKey(channelID)) {
+            linkedChannels.put(channelID, true);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        stopScheduler();
+    }
+
+    private void stopScheduler() {
+        ScheduledFuture<?> job = refreshJob;
+        if (job != null) {
+            job.cancel(true);
+        }
+        refreshJob = null;
+    }
+
+    /**
+     * Start the job refreshing the oven status
+     */
+    private void startAutomaticRefresh() {
+        ScheduledFuture<?> job = refreshJob;
+        if (job == null || job.isCancelled()) {
+            int period = config.refreshInterval;
+            refreshJob = scheduler.scheduleWithFixedDelay(this::run, 0, period, TimeUnit.SECONDS);
+        }
+    }
+
+    private void run() {
```

Comment:
Maybe find a better name for your method, like for example `refreshChannels`

Reply:
ok